File: php/features.php
<?php
declare(strict_types=1);

// KittyNetVpn SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class KittyNetVpnFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new KittyNetVpnBaseFeature();
            case "test":
                return new KittyNetVpnTestFeature();
            default:
                return new KittyNetVpnBaseFeature();
        }
    }
}

File: rb/features.rb
# KittyNetVpn SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module KittyNetVpnFeatures
  def self.make_feature(name)
    case name
    when "base"
      KittyNetVpnBaseFeature.new
    when "ratelimit"
      KittyNetVpnRatelimitFeature.new
    when "retry"
      KittyNetVpnRetryFeature.new
    when "test"
      KittyNetVpnTestFeature.new
    when "timeout"
      KittyNetVpnTimeoutFeature.new
    else
      KittyNetVpnBaseFeature.new
    end
  end
end
